python/plugins/processing/algs/qgis/TilesXYZ.py

```diff
@@ -0,0 +1,366 @@
+            max(-89.98, self.extent[1] - (first_tile_extent[3] - first_tile_extent[1])),
+            # extend with width of 1 tile, do not exceed 180
+            min(180, self.extent[2] + (first_tile_extent[2] - first_tile_extent[0])),
+            first_tile_extent[3]
+        ]
+        bounds = ','.join(map(str, zoom_extent))
+        ds.ExecuteSQL("UPDATE metadata SET value='{}' WHERE name='bounds'".format(bounds))
+        ds = None
+
+        self._zoomDs = gdal.OpenEx(self.filename, 1, open_options=['ZOOM_LEVEL=%s' % first_tile.z])
+        self._first_tile = first_tile
+        self._zoom = first_tile.z
+
+    def writeTile(self, tile, image):
+        if tile.z != self._zoom:
+            self._initZoomLayer(tile)
+
+        data = QByteArray()
+        buff = QBuffer(data)
+        image.save(buff, 'PNG')
+
+        mmap_name = '/vsimem/' + uuid4().hex
+        gdal.FileFromMemBuffer(mmap_name, data.data())
+        gdal_dataset = gdal.Open(mmap_name)
+        data = gdal_dataset.ReadRaster(0, 0, self.tile_width, self.tile_height)
+        gdal_dataset = None
+        gdal.Unlink(mmap_name)
+
+        xoff = (tile.x - self._first_tile.x) * self.tile_width
+        yoff = (tile.y - self._first_tile.y) * self.tile_height
+        self._zoomDs.WriteRaster(xoff, yoff, self.tile_width, self.tile_height, data)
+
+    def close(self):
+        self._zoomDs = None
+        sqlite_driver = ogr.GetDriverByName('SQLite')
+        ds = sqlite_driver.Open(self.filename, 1)
+        bounds = ','.join(map(str, self.extent))
+        ds.ExecuteSQL("UPDATE metadata SET value='{}' WHERE name='bounds'".format(bounds))
+        ds = None
+        return self.filename
+
+
+class TilesXYZ(QgisAlgorithm):
+    NAME = 'NAME'
+    EXTENT = 'EXTENT'
+    ZOOM_MIN = 'ZOOM_MIN'
+    ZOOM_MAX = 'ZOOM_MAX'
+    TILE_FORMAT = 'TILE_FORMAT'
+    DPI = 'DPI'
+    OUTPUT_FORMAT = 'OUTPUT_FORMAT'
+    OUTPUT_DIRECTORY = 'OUTPUT_DIRECTORY'
+
+    def group(self):
+        return self.tr('Raster tools')
+
+    def groupId(self):
+        return 'rastertools'
+
+    def initAlgorithm(self, config=None):
+        self.addParameter(QgsProcessingParameterString(self.NAME, self.tr('Name'), defaultValue='Test'))
+        self.addParameter(QgsProcessingParameterFolderDestination(self.OUTPUT_DIRECTORY, self.tr('Output directory')))
+
+        self.addParameter(QgsProcessingParameterExtent(self.EXTENT, self.tr('Extent')))
+        self.addParameter(QgsProcessingParameterNumber(self.ZOOM_MIN,
+                                                       self.tr('Minimum zoom'),
+                                                       defaultValue=12))
+        self.addParameter(QgsProcessingParameterNumber(self.ZOOM_MAX,
+                                                       self.tr('Maximum zoom'),
+                                                       defaultValue=12))
+        self.addParameter(QgsProcessingParameterNumber(self.DPI, self.tr('DPI'), defaultValue=96))
```

Comment:
Same here -- a dpi <1 should not be permitted

Reply:
All numeric parameters should now have a reasonable min/max values set.